packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -0,0 +1,327 @@
+  }
+
+  Stream<List<PurchaseDetails>> get purchaseStream =>
+      _purchaseUpdatedController.stream;
+  static late StreamController<List<PurchaseDetails>>
+      _purchaseUpdatedController;
+
+  /// The [BillingClient] that's abstracted by [GooglePlayConnection].
+  ///
+  /// This field should not be used out of test code.
+  @visibleForTesting
+  late final BillingClient billingClient;
+
+  late Future<void> _readyFuture;
+  static Set<String> _productIdsToConsume = Set<String>();
+
+  @override
+  Future<bool> isAvailable() async {
+    await _readyFuture;
+    return billingClient.isReady();
+  }
+
+  @override
+  Future<bool> buyNonConsumable({required PurchaseParam purchaseParam}) async {
+    if (!(purchaseParam is GooglePlayPurchaseParam)) {
+      throw ArgumentError(
+        'On Android, the `purchaseParam` should always be of type `GooglePlayPurchaseParam`.',
+      );
+    }
+
+    BillingResultWrapper billingResultWrapper =
+        await billingClient.launchBillingFlow(
+            sku: purchaseParam.productDetails.id,
+            accountId: purchaseParam.applicationUserName,
+            oldSku: purchaseParam
+                .changeSubscriptionParam?.oldPurchaseDetails.productID,
+            purchaseToken: purchaseParam.changeSubscriptionParam
+                ?.oldPurchaseDetails.verificationData.serverVerificationData,
+            prorationMode:
+                purchaseParam.changeSubscriptionParam?.prorationMode);
+    return billingResultWrapper.responseCode == BillingResponse.ok;
+  }
+
+  @override
+  Future<bool> buyConsumable(
+      {required PurchaseParam purchaseParam, bool autoConsume = true}) {
+    if (autoConsume) {
+      _productIdsToConsume.add(purchaseParam.productDetails.id);
+    }
+    return buyNonConsumable(purchaseParam: purchaseParam);
+  }
+
+  @override
+  Future<BillingResultWrapper> completePurchase(
+      PurchaseDetails purchase) async {
+    assert(
+      purchase is GooglePlayPurchaseDetails,
+      'On Android, the `purchase` should always be of type `GooglePlayPurchaseDetails`.',
+    );
+
+    GooglePlayPurchaseDetails googlePurchase =
+        purchase as GooglePlayPurchaseDetails;
+
+    if (googlePurchase.billingClientPurchase.isAcknowledged) {
+      return BillingResultWrapper(responseCode: BillingResponse.ok);
+    }
+
+    if (googlePurchase.verificationData == null) {
+      throw ArgumentError(
+          'completePurchase unsuccessful. The `purchase.verificationData` is not valid');
+    }
+
+    return await billingClient
+        .acknowledgePurchase(purchase.verificationData.serverVerificationData);
+  }
+
+  /// Mark that the user has consumed a product.
+  ///
+  /// You are responsible for consuming all consumable purchases once they are
+  /// delivered. The user won't be able to buy the same product again until the
+  /// purchase of the product is consumed.
+  Future<BillingResultWrapper> consumePurchase(PurchaseDetails purchase) {
```

Comment:
This should also be in the addition class

Reply:
Done.